nfe/src/base/dest.rs

```diff
@@ -33,9 +34,9 @@ pub enum IndicadorContribuicaoIe {
 }
 
 impl FromStr for Destinatario {
-    type Err = String;
+    type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        serde_xml_rs::from_str(s).map_err(|e| e.to_string())
+        serde_xml_rs::from_str(s).map_err(|e| e.into())
```

Comment:
serde_xml_rs::from_str(s)

deve funcionar.

Reply:
Eu já até tinha tentado usar o `from_str` sem esse `map_err`:
```
error[E0308]: mismatched types
  --> src/base/dest.rs:40:9
   |
39 |     fn from_str(s: &str) -> Result<Self, Self::Err> {
   |                             ----------------------- expected `Result<base::dest::Destinatario, base::error::Error>` because of return type
40 |         serde_xml_rs::from_str(s)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `base::error::Error`, found enum `serde_xml_rs::Error`
   |
   = note: expected enum `Result<base::dest::Destinatario, base::error::Error>`
              found enum `Result<_, serde_xml_rs::Error>`

error: aborting due to previous error
```

Creio que eu precisaria implementar algum outro `From`. No momento não vejo problema em seguir com o uso do `map_err`